content/300-guides/050-database/500-using-prisma-with-planetscale.mdx

```diff
@@ -0,0 +1,181 @@
+
+## How to enable emulation of referential integrity
+
+Referential integrity is currently a Preview feature. To enable this, add it to the `previewFeatures` list in the `generator` block of `schema.prisma`, and set the `referentialIntegrity` type to `"prisma"` in the `datasource` block:
+
+```prisma file=schema.prisma
+datasource db {
+  provider             = "mysql"
+  url                  = env("DATABASE_URL")
+  referentialIntegrity = "prisma"
+}
+
+generator js {
+  provider        = "prisma-client-js"
+  previewFeatures = ["referentialIntegrity"]
+}
+```
+
+## How to use branches and deploy requests
+
+When connecting to PlanetScale with Prisma, you will need to use the correct connection string for your branch. The connection URL for a given database branch can be found from your PlanetScale account by going to the overview page for the branch and selecting the 'Connect' dropdown. In the 'Passwords' section, generate a new password and select 'Prisma' from the dropdown to get the Prisma format for the connection URL.
+
+See Prisma's [Getting Started guide](/getting-started/setup-prisma/start-from-scratch/relational-databases/connect-your-database-typescript-planetscale) for more details of how to connect to a PlanetScale database.
+
+## How to create indexes on foreign keys
+
+As an example of a situation where you would want to an add an index, take this schema for a blog with posts and comments:
+
+```prisma file=schema.prisma
+model Post {
+  id       Int       @id @default(autoincrement())
+  title    String
+  content  String
+  likes    Int       @default(0)
+  comments Comment[]
+}
+
+model Comment {
+  id      Int    @id @default(autoincrement())
+  comment String
+  postId  Int
+  post    Post   @relation(fields: [postId], references: [id], onDelete: Cascade)
+}
+```
+
+The `postId` field in the `User` model refers to the corresponding `id` field in the `Post` model. However this is not implemented as a foreign key in PlanetScale, so it doesn't have an index. This means that some queries may not be well optimised. For example, if you query for all comments with a certain post `id`, PlanetScale may have to do a full table lookup, which could be slow and expensive.
+
+To avoid this, you can define an index on the `postId` field using [Prisma's `@@index` argument](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#index):
+
+```prisma file=schema.prisma highlight=15;add
+model Post {
+  id       Int       @id @default(autoincrement())
+  title    String
+  content  String
+  likes    Int       @default(0)
+  comments Comment[]
+}
+
+model Comment {
+  id      Int    @id @default(autoincrement())
+  comment String
+  postId  Int
+  post    Post   @relation(fields: [postId], references: [id], onDelete: Cascade)
+
+  @@index([postId])
+}
+```
+
+You can then add this change to your schema [using `db push`](#how-to-make-schema-changes-with-db-push).
+
+## How to make schema changes with `db push`
```

Comment:
In this section, we should recommend first creating a new database branch on PlanetScale. 

Additionally, I would suggest adding a new column/field rather than renaming `likes`. 

Because renames are not supported in PlanetScale, this operation leads to data loss which is obviously undesired. While it's possible to do renames with the expand and contract pattern, it requires multiple steps and adjustments to application code; so I think we should go for the more common schema migration adding a new column.

Reply:
Thanks, will do this. The rename was a bit of an editing mixup after first trying to include the expand and contract pattern and then taking it out again. I'll rewrite with just adding a field instead.

Reply:
Done.